### PR TITLE
[PAY-1514] Mobile USDC track screen changes

### PR DESCRIPTION
**Pull request by dharit-tan**
### Description
Mobile track screen changes for USDC content

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

Local ios stage
Simulator Screenshot - iPhone 14 Pro - 2023-07-11 at 17 00 24
Simulator Screenshot - iPhone 14 Pro - 2023-07-11 at 17 00 16
Simulator Screenshot - iPhone 14 Pro - 2023-07-11 at 17 01 14

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rt-usdc-track-screen

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rt-usdc-track-screen

File: packages/common/src/utils/formatUtil.ts [hunk 1]
```diff
@@ -180,6 +181,12 @@ export const parseWeiNumber = (number: string) => {
   }
 }
 
+// TODO convert to new fn
+export const formatUSDC = (number: StringUSDC) => {
```
Comment (schottra) [inline packages/common/src/utils/formatUtil.ts:185]: This is merged now if you'd like to pull in latest from the feature branch and use it. The function is in `utils/wallet.ts`

File: packages/mobile/src/components/core/LockedStatusBadge.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+/** Renders a small badge with locked or unlocked icon */
+export const LockedStatusBadge = ({
+  locked,
+  variant
+}: LockedStatusBadgeProps) => {
+  const styles = useStyles()
+  const staticWhite = useColor('staticWhite')
+  const LockComponent = locked ? IconLock : IconLockUnlocked
+  return (
+    <View style={[styles.root, variant === 'premium' ? styles.premium : null]}>
```
Comment (schottra) [inline packages/mobile/src/components/core/LockedStatusBadge.tsx:33]: per the conversation at standup, want to change the `premium` stuff to `purchase`?

File: packages/mobile/src/components/details-tile/DetailsTileHasAccess.tsx [hunk 3]
```diff
@@ -1,24 +1,31 @@
...
 import { useNavigation } from 'app/hooks/useNavigation'
 import { flexRowCentered, makeStyles } from 'app/styles'
-import { useColor, useThemePalette } from 'app/utils/theme'
+import { useColor } from 'app/utils/theme'
 
 const messages = {
   unlocked: 'UNLOCKED',
   collectibleGated: 'COLLECTIBLE GATED',
   specialAccess: 'SPECIAL ACCESS',
+  USDCPurchaseTitle: 'Pay to Unlock',
```
Comment (schottra) [inline packages/mobile/src/components/details-tile/DetailsTileHasAccess.tsx:28]: nit: can probably remove usdc reference from this variable to make things easier? (`payToUnlock: 'Pay to Unlock'`)

File: packages/mobile/src/components/details-tile/DetailsTileNoAccess.tsx [hunk 4]
```diff
@@ -322,20 +339,53 @@ export const DetailsTileNoAccess = ({
...
+      return (
+        <>
+          <View style={styles.descriptionContainer}>
+            <Text style={styles.description}>
+              {messages.lockedUSDCPurchase}
+            </Text>
+          </View>
+          <Button
+            style={[styles.mainButton, styles.buyButton]}
+            styles={{ icon: { width: 16, height: 16 } }}
```
Comment (schottra) [inline packages/mobile/src/components/details-tile/DetailsTileNoAccess.tsx:352]: spacing variables?

File: packages/mobile/src/components/lineup-tile/LineupTileStats.tsx [hunk 5]
```diff
@@ -205,24 +192,14 @@ export const LineupTileStats = ({
...
-          ]}
-        >
-          {doesUserHaveAccess ? (
-            <IconUnlocked fill={staticWhite} height={14} width={14} />
-          ) : (
-            <IconLock fill={staticWhite} height={14} width={14} />
-          )}
-        </View>
+        <LockedStatusBadge
+          locked={!doesUserHaveAccess ?? true}
```
Comment (schottra) [inline packages/mobile/src/components/lineup-tile/LineupTileStats.tsx:196]: ?

File: packages/mobile/src/components/lineup-tile/LineupTileStats.tsx [hunk 6]
```diff
@@ -17,17 +17,16 @@ import { View, TouchableOpacity } from 'react-native'
...
-import IconUnlocked from 'app/assets/images/iconUnlocked.svg'
 import { CollectionDownloadStatusIndicator } from 'app/components/offline-downloads/CollectionDownloadStatusIndicator'
 import { TrackDownloadStatusIndicator } from 'app/components/offline-downloads/TrackDownloadStatusIndicator'
 import Text from 'app/components/text'
-import { useIsUSDCEnabled } from 'app/hooks/useIsUSDCEnabled'
 import { useNavigation } from 'app/hooks/useNavigation'
 import { makeStyles, flexRowCentered } from 'app/styles'
 import { useThemeColors } from 'app/utils/theme'
 
+import { LockedStatusBadge } from '../core'
```
[the remark this answers is not in the document]
Comment (dharit-tan) [reply packages/mobile/src/components/lineup-tile/LineupTileStats.tsx:28]: oh oops this was auto-import

File: packages/mobile/src/components/lineup-tile/LineupTileStats.tsx (hunk 5 above)
Comment (dharit-tan) [reply packages/mobile/src/components/lineup-tile/LineupTileStats.tsx:196]: `doesUserHaveAccess` could be undefined. default to true seems better than false?

Comment (schottra) [reply packages/mobile/src/components/lineup-tile/LineupTileStats.tsx:196]: ! is going to convert the value to a boolean. Anything falsey (undefined, null, 0, an empty string) will result in `locked` being `true`. So I think null coalescing to `true` isn't necessary here?
At first glance, I thought this was a temporary thing you were doing for testing.

File: packages/mobile/src/components/details-tile/DetailsTile.tsx [hunk 7]
```diff
@@ -229,9 +206,12 @@ export const DetailsTile = ({
   })
 
   const isOwner = user?.user_id === currentUserId
+  // const isOwner = true
```
Comment (schottra) [inline packages/mobile/src/components/details-tile/DetailsTile.tsx:209]: remove?

Comment (dharit-tan) [reply packages/mobile/src/components/details-tile/DetailsTile.tsx:209]: lol ty

**Pull request merged**